class.md recreated (#1719)
Codecademy/docs: content/html/concepts/attributes/terms/class/class.md

---

> Title: 'Classes'

Reviewer: ```suggestion
Title: 'class'
```

> Description: 'Classes are HTML attributes used to select one or more elements for the application of styling or logic.'

Reviewer: ```suggestion
Description: 'class is a HTML attribute used to select one or more elements for the application of styling or logic.'
```

Subjects:
  - 'Web Development'
  - 'Web Design'

Tags:
  - 'Attributes'
  - 'Classes'

>   - 'Style'

Reviewer: The end of the metadata is missing. There needs to be a CatalogContent: section and a concluding three dashes: `---`

Also, the extra line breaks in the metadata need to be removed.

The complete Metadata should look like this:
```
---
Title: 'The Title'
Description: 'A brief description'
Subjects:
  - 'A subject name'
  - 'A second subject name'
  - 'An nth subject name'
Tags:
  - 'A tag'
  - 'A second tag'
  - 'An nth tag'
CatalogContent:
  - 'learn-example-course'
  - 'path/example-path'
---
```

You can find entries for CatalogContent: [here](https://github.com/Codecademy/docs/blob/main/documentation/catalog-content.md).

> Classes are HTML attributes used to select one or more elements for CSS styling or JavaScript purposes.

Reviewer: ```suggestion
**`class`** is a HTML attribute used to select one or more elements for CSS styling or JavaScript purposes.
```

## For Styling

The `class` attribute is used for assigning styles to elements. Multiple classes can be assigned to a single element in a space-separated list. (e.g. `class="class1 class2 class3"`)

Inside the stylesheet, a class name is selected with the following syntax:

```css
.class-1 {
}
```

A class selector is a name preceded by a period (`.`).

> Below is some markup with a `title`, containing a `header-1` and `P1` and  `P2`. The styling for the `header-1 `, elements with the `P1` and `P1`class name received different styling.

Reviewer: ```suggestion
Below is some markup with the classes `header-1` and `P1` and  `P2`. Elements with the `header-1`, `P1` and `P2` class names receive different styling.
```

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <style>
      .header-1 {
        background-color: lightblue;
        font-size: 40px;
      }

      .P1 {
        background-color: white;
        font-size: 25px;
      }
      .P2 {
        background-color: blue;
        font-size: 20px;
      }
    </style>
    <title>Classes</title>
  </head>
  <body>
  <!--HEADER-->
    
    <p><img src="images/android-chrome-512x512.png" width="360px" height="240px"></p>
    
    <div class="title">
    <h1 class="header-1">PIZZA MAN HE DELIVERS</h1>    
  </div>
  <!--HISTORY-->

    <p class="P1">Pizza Man he delivers is a family-owned business since 1964. Our goal is to make the best pizza and have it delivered to your place.</p>
    <p class="P2">We have 3 locations so far and growing. </p>
    </body>
 
</html>
```

## For JavaScript

> The `class` attribute is used to perform JavaScript operations on HTML elements. Through the Document Object Model ("DOM"), there are various JavaScript methods built for selecting elements via their class.

Reviewer: ```suggestion
The `class` attribute is used to perform JavaScript operations on HTML elements. Through the Document Object Model ("DOM") there are various JavaScript methods built for selecting elements via their class.
```

Assuming the markup from the last example is linked to a JavaScript file:

```js
const innerDiv = document
  .getElementsByClassName('header-1')
  console.log(innerDiv)

> 

Reviewer: Remove the extra line breaks

```

> The code above is getting an element with a class property and prints it to the console

Reviewer: ```suggestion
The code above gets an element with a class property and prints it to the console.
```

> recreated on 02/24/2023 again

Reviewer: ```suggestion
```
Remove this line.

Inside the stylesheet, a class name is selected with the following syntax:

```css
.class-1 {
}
```

A class selector is a name preceded by a period (`.`).

Below is some markup with the classes `header-1` and `P1` and `P2`. Elements with the `header-1`, `P1` and `P2` class names receive different styling.

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <style>
      .header-1 {
        background-color: lightblue;
        font-size: 40px;
      }
      .P1 {
        background-color: white;
        font-size: 25px;
      }
      .P2 {
        background-color: blue;
        font-size: 20px;
      }
    </style>
    <title>Classes</title>
  </head>
  <body>
    <!--HEADER-->
    <p>
      <img
        src="images/android-chrome-512x512.png"
        width="360px"
        height="240px"
      />
    </p>

    <div class="title">
      <h1 class="header-1">PIZZA MAN</h1>
    </div>
    <!--HISTORY-->
    <p class="P1">
      Pizza Man is a family-owned business since 1964. Our goal is to make the
      best pizza and have it delivered to your place.
    </p>
    <p class="P2">We have 3 locations so far and growing.</p>
  </body>
</html>
```

## For JavaScript

> The `class` attribute is used to perform JavaScript operations on HTML elements. Through the Document Object Model ("DOM") there are various JavaScript methods built for selecting elements via their class.

Reviewer: ```suggestion
The `class` attribute is used to perform [JavaScript](https://www.codecademy.com/resources/docs/javascript) operations on HTML elements. Through the Document Object Model ("DOM") there are various JavaScript methods built for selecting elements via their class.
```

> ---

Reviewer: You still need to fix the metadata. You have extra line breaks between sections and you're missing the concluding three dashes: `---`

Again, it should look like this:

```
---
Title: 'The Title'
Description: 'A brief description'
Subjects:
  - 'A subject name'
  - 'A second subject name'
  - 'An nth subject name'
Tags:
  - 'A tag'
  - 'A second tag'
  - 'An nth tag'
CatalogContent:
  - 'learn-example-course'
  - 'path/example-path'
---
```

<html lang="en">
  <head>
    <style>
      .header-1 {
        background-color: lightblue;
        font-size: 40px;
      }

      .P1 {
        background-color: white;
        font-size: 25px;
      }
      .P2 {
        background-color: blue;
        font-size: 20px;
      }
    </style>
    <title>Classes</title>
  </head>
  <body>
  <!--HEADER-->
    
    <p><img src="images/android-chrome-512x512.png" width="360px" height="240px"></p>
    
    <div class="title">
    <h1 class="header-1">PIZZA MAN HE DELIVERS</h1>    
  </div>
  <!--HISTORY-->

    <p class="P1">Pizza Man he delivers is a family-owned business since 1964. Our goal is to make the best pizza and have it delivered to your place.</p>
    <p class="P2">We have 3 locations so far and growing. </p>
    </body>
 
</html>
```

## For JavaScript

The `class` attribute is used to perform JavaScript operations on HTML elements. Through the Document Object Model ("DOM") there are various JavaScript methods built for selecting elements via their class.

Assuming the markup from the last example is linked to a JavaScript file:

```js
const innerDiv = document
  .getElementsByClassName('header-1')
  console.log(innerDiv)

> 

Reviewer: You still need to remove the extra line breaks in the code block.
